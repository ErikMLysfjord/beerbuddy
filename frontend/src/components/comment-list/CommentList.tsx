import CommentItem from "../comment-item/CommentItem";
import styles from "./CommentList.module.css";

const CommentList = () => (
  <section>
    <div className={styles.container}>
      <button className={styles.commentButton}>Comment</button>
    </div>
    <hr className={styles.divider} />
    <div className={styles.commentListContainer}>
      <ul className={styles.commentList}>
        <li>
          <CommentItem />
        </li>
        <li>
          <CommentItem />
        </li>
        <li>
          <CommentItem />
        </li>
        <li>
          <CommentItem />
        </li>
        <li>
          <CommentItem />
        </li>
        <li>
          <CommentItem />
        </li>
      </ul>
    </div>
  </section>
);

export default CommentList;
