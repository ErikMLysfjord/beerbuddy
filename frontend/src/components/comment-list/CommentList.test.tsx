import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import CommentList from "./CommentList";

describe("CommentList", () => {
  it("matches snapshot", () => {
    const { container } = render(<CommentList />);
    expect(container).toMatchSnapshot();
  });

  it("renders correct amount of comments", () => {
    const { getAllByRole } = render(<CommentList />);
    expect(getAllByRole("listitem").length).toBe(6);
  });
});
