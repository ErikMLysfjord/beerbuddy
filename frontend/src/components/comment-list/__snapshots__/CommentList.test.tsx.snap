// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentList > matches snapshot 1`] = `
<div>
  <section>
    <div
      class="_container_907aa3"
    >
      <button
        class="_commentButton_907aa3"
      >
        Comment
      </button>
    </div>
    <hr
      class="_divider_907aa3"
    />
    <div
      class="_commentListContainer_907aa3"
    >
      <ul
        class="_commentList_907aa3"
      >
        <li>
          <div
            class="_commentContainer_7a79ad"
          >
            <div
              class="_emptyProfilePic_7a79ad"
            />
            <div
              class="_commentContentContainer_7a79ad"
            >
              <div>
                <div
                  class="_commentHeader_7a79ad"
                >
                  <p
                    class="_commentAuthor_7a79ad"
                  >
                    Martin Hansa-Borg
                  </p>
                  <p
                    class="_dot_7a79ad"
                  >
                    •
                  </p>
                  <p
                    class="_commentDate_7a79ad"
                  >
                    5 days ago
                  </p>
                </div>
              </div>
              <p
                class="_commentText_7a79ad"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla cursus sem diam, eget consectetur quam interdum eu. Aliquam ornare lectus vel sapien blandit eleifend. Aenean ac vestibulum metus. Nam auctor sed dui dapibus pharetra.
              </p>
            </div>
          </div>
        </li>
        <li>
          <div
            class="_commentContainer_7a79ad"
          >
            <div
              class="_emptyProfilePic_7a79ad"
            />
            <div
              class="_commentContentContainer_7a79ad"
            >
              <div>
                <div
                  class="_commentHeader_7a79ad"
                >
                  <p
                    class="_commentAuthor_7a79ad"
                  >
                    Martin Hansa-Borg
                  </p>
                  <p
                    class="_dot_7a79ad"
                  >
                    •
                  </p>
                  <p
                    class="_commentDate_7a79ad"
                  >
                    5 days ago
                  </p>
                </div>
              </div>
              <p
                class="_commentText_7a79ad"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla cursus sem diam, eget consectetur quam interdum eu. Aliquam ornare lectus vel sapien blandit eleifend. Aenean ac vestibulum metus. Nam auctor sed dui dapibus pharetra.
              </p>
            </div>
          </div>
        </li>
        <li>
          <div
            class="_commentContainer_7a79ad"
          >
            <div
              class="_emptyProfilePic_7a79ad"
            />
            <div
              class="_commentContentContainer_7a79ad"
            >
              <div>
                <div
                  class="_commentHeader_7a79ad"
                >
                  <p
                    class="_commentAuthor_7a79ad"
                  >
                    Martin Hansa-Borg
                  </p>
                  <p
                    class="_dot_7a79ad"
                  >
                    •
                  </p>
                  <p
                    class="_commentDate_7a79ad"
                  >
                    5 days ago
                  </p>
                </div>
              </div>
              <p
                class="_commentText_7a79ad"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla cursus sem diam, eget consectetur quam interdum eu. Aliquam ornare lectus vel sapien blandit eleifend. Aenean ac vestibulum metus. Nam auctor sed dui dapibus pharetra.
              </p>
            </div>
          </div>
        </li>
        <li>
          <div
            class="_commentContainer_7a79ad"
          >
            <div
              class="_emptyProfilePic_7a79ad"
            />
            <div
              class="_commentContentContainer_7a79ad"
            >
              <div>
                <div
                  class="_commentHeader_7a79ad"
                >
                  <p
                    class="_commentAuthor_7a79ad"
                  >
                    Martin Hansa-Borg
                  </p>
                  <p
                    class="_dot_7a79ad"
                  >
                    •
                  </p>
                  <p
                    class="_commentDate_7a79ad"
                  >
                    5 days ago
                  </p>
                </div>
              </div>
              <p
                class="_commentText_7a79ad"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla cursus sem diam, eget consectetur quam interdum eu. Aliquam ornare lectus vel sapien blandit eleifend. Aenean ac vestibulum metus. Nam auctor sed dui dapibus pharetra.
              </p>
            </div>
          </div>
        </li>
        <li>
          <div
            class="_commentContainer_7a79ad"
          >
            <div
              class="_emptyProfilePic_7a79ad"
            />
            <div
              class="_commentContentContainer_7a79ad"
            >
              <div>
                <div
                  class="_commentHeader_7a79ad"
                >
                  <p
                    class="_commentAuthor_7a79ad"
                  >
                    Martin Hansa-Borg
                  </p>
                  <p
                    class="_dot_7a79ad"
                  >
                    •
                  </p>
                  <p
                    class="_commentDate_7a79ad"
                  >
                    5 days ago
                  </p>
                </div>
              </div>
              <p
                class="_commentText_7a79ad"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla cursus sem diam, eget consectetur quam interdum eu. Aliquam ornare lectus vel sapien blandit eleifend. Aenean ac vestibulum metus. Nam auctor sed dui dapibus pharetra.
              </p>
            </div>
          </div>
        </li>
        <li>
          <div
            class="_commentContainer_7a79ad"
          >
            <div
              class="_emptyProfilePic_7a79ad"
            />
            <div
              class="_commentContentContainer_7a79ad"
            >
              <div>
                <div
                  class="_commentHeader_7a79ad"
                >
                  <p
                    class="_commentAuthor_7a79ad"
                  >
                    Martin Hansa-Borg
                  </p>
                  <p
                    class="_dot_7a79ad"
                  >
                    •
                  </p>
                  <p
                    class="_commentDate_7a79ad"
                  >
                    5 days ago
                  </p>
                </div>
              </div>
              <p
                class="_commentText_7a79ad"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla cursus sem diam, eget consectetur quam interdum eu. Aliquam ornare lectus vel sapien blandit eleifend. Aenean ac vestibulum metus. Nam auctor sed dui dapibus pharetra.
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
